routing/router.go

```diff
@@ -817,25 +822,47 @@ func (r *ChannelRouter) pruneZombieChans() error {
 			info.ChannelID)
 
 		// TODO(roasbeef): add ability to delete single directional edge
-		chansToPrune = append(chansToPrune, info.ChannelID)
+		chansToPrune[info.ChannelID] = struct{}{}
 
 		return nil
 	}
 
-	err := r.cfg.Graph.ForEachChannel(filterPruneChans)
+	startTime := time.Unix(0, 0)
+	endTime := time.Now().Add(-1 * chanExpiry)
+	oldEdges, err := r.cfg.Graph.ChanUpdatesInHorizon(startTime, endTime)
+	if err != nil {
+		return fmt.Errorf("unable to filter local zombie "+
+			"chans: %v", err)
+	}
+
+	disabledChanIDs, err := r.cfg.Graph.DisabledChannelIDs()
```

Comment:
Looking at this again, we should only be calling this when `AssumeChannelValid` is set. We can also remove the double disabled logic within the filtering closure above and avoid filtering these as we already know they're double disabled.

Reply:
@wpaulino it will be nice to be able to populate the disabledChanIDs straight into the map without loading the ChannelEdge from the db and passing it through the filtering closure.
One caveat though is that there is a check there to skip self channels. Any idea on how to do that filtering with the channelID alone? Or if that filtering is needed at the disabled case at all?

Reply:
We could provide an optional compressed public key to `DisabledChannelIDs` and do the filtering there.

Reply:
@wpaulino it seems that loading the ChannelEdgeInfo is inevitable in order to get the pubkey for filtering, so I added a small helper function `isSelfChannelEdge` and used it to filter the disabled channels only if AssumeChannelValid=true.
Did this in a separate commit so you can easily review. Once the review is finished I will sqash.
Also rebased.